### PR TITLE
Add Custom Color Support, bringing 12 new Colors

### PR DESCRIPTION
**Pull request by EoF-1141**
- Added CustomColors, to add 12 new Colors. ( Requested in #77 )
- Modified Helpers.isLighterColor, to add support for Medic, Hacker etc. using Darker/Lighter.
- Modified ColorChecker, to only allow picking duplicates if there are more players than unique colors.

### CODE REVIEW

File: Source Code/CrowdedPatches.cs [hunk 1]
```diff
@@ -144,7 +144,8 @@ public static class GameDataGetAvailableIdPatch {
     [HarmonyPatch(typeof(PlayerControl), nameof(PlayerControl.CmdCheckColor))]
     public static class PlayerControlCmdCheckColorPatch {
         public static bool Prefix(PlayerControl __instance, byte CKNNKEMNHAK) {
-            if (!Helpers.isCustomServer()) return true;
+            if (!Helpers.isCustomServer() || BLMBFIODBKL.AEDCMKGJKAG.Count >= PlayerControl.AllPlayerControls.Count) // Palette.PlayerColors
+                return true;
 
             MessageWriter writer = AmongUsClient.Instance.StartRpcImmediately(PlayerControl.LocalPlayer.NetId, (byte)CustomRPC.SetUncheckedColor, Hazel.SendOption.Reliable, -1);
```
Comment (Eisbison) [inline Source Code/CrowdedPatches.cs:150]: :+1:

File: Source Code/Helpers.cs [hunk 2]
```diff
@@ -164,8 +164,10 @@ public static PlayerControl playerById(byte id)
             }
         }
 
-        private static List<int> lighterColors = new List<int>(){ 3, 4, 5, 7, 10, 11};
+        private static List<int> lighterColors = new List<int>(){ 3, 4, 5, 7, 10, 11 };
```
Comment (Eisbison) [inline Source Code/Helpers.cs:167]: Should we add these ids to the CustomColors.lighterColors?

**Review by Eisbison — COMMENTED** [review]
The code looks good to me, I'll do some testing tomorrow :+1:

File: Source Code/Helpers.cs (hunk 2 above)
Comment (EoF-1141) [reply Source Code/Helpers.cs:167]: Not sure if it should go into CustomColors, as they are default colors.
However could revert changes to Helpers.cs, and use Reflection in CustomColors, to have a shared Reference to "lighterColors" between Helpers and CustomColors

**Pull request closed without merging**